keps/sig-storage/284-enable-volume-expansion/README.md

```diff
@@ -344,3 +355,229 @@ type StorageClass struct {
 
 This proposal relies on ability to update PVC status from kubelet. While updating PVC's status
 a PATCH request must be made from kubelet to update the status.
+
+## Production Readiness Review Questionnaire
+
+### Feature Enablement and Rollback
+
+###### How can this feature be enabled / disabled in a live cluster?
+
+Volume expansion as a feature has been in beta for too long and as a result has gathered
+different feature gates that control various aspects of expansion.
+
+- [x] Feature gate (also fill in values in `kep.yaml`)
+  - Feature gate name: ExpandPersistentVolumes
+    - description: Parent feature gate that is required for expansion in general to work.
+    - Components depending on the feature gate:
+        - kube-apiserver
+        - kubelet
+        - kube-controller-manager
+  - Feature gate name: ExpandInUsePersistentVolumes
+    - description: Enables online expansion. Requires ExpandPersistentVolumes feature gate.
+    - Components depending on the feature gate:
+        - kube-apiserver
+        - kubelet
+        - kube-controller-manager
+  - Feature gate name: ExpandCSIVolumes
+    - description: Enables CSI expansion.
+    - Components depending on the feature gate:
+        - kube-apiserver
+        - kubelet
+        - kube-controller-manager
+- [ ] Other
+  - Describe the mechanism:
+  - Will enabling / disabling the feature require downtime of the control
+    plane?
+  - Will enabling / disabling the feature require downtime or reprovisioning
+    of a node? (Do not assume `Dynamic Kubelet Config` feature is enabled).
+
+###### Does enabling the feature change any default behavior?
+
+Enabling the feature gate allows users to increase size of pvc by editing `pvc.Spec.Resources` which results
+in Kubernetes trying to full-fill the request by actually expanding the volume in controller and then performing
+file system or any other kind of expansion needed on the node.
+
+###### Can the feature be disabled once it has been enabled (i.e. can we roll back the enablement)?
+
+Yes - it can be disabled. It just means users can no longer expand their PVCs.
+
+###### What happens if we reenable the feature if it was previously rolled back?
+
+It should be safe to do that. It will just re-enable the feature.
+
+###### Are there any tests for feature enablement/disablement?
+
+There aren't any e2e but there are unit tests that cover this behaviour.
+
+### Rollout, Upgrade and Rollback Planning
+
+###### How can a rollout or rollback fail? Can it impact already running workloads?
+
+The feature gate should not impact existing workloads but since we try to expand the
+file system(or perform node-expansion) during volume mount and if expansion fails with
+some kind of terminal error then it may prevent mount operation from succeeding.
+
+###### What specific metrics should inform a rollback?
+
+The `volume_mount` operation failure metric - `storage_operation_errors_total{operation_name=volume_mount}`
```

Comment:
What about controller and CSI failures? Do we have failure metrics that's more fine-grained to just the  expand operation?

Reply:
storage_operation_errors_total{operation_name=expand_volume} and storage_operation_errors_total{operation_name=volume_fs_resize}?

Reply:
Currently we do not any specific metrics from external-resizer that are more fine grained. We will have to add those I think.

Reply:
Added some text about adding new metric from external-resizer.

Reply:
How come https://github.com/kubernetes-csi/external-resizer/pull/67 does not suffice?

Reply:
hmm, I did originally document `csi_sidecar_operations_seconds` - but was wondering if we still need intree equivalent but I have come to the conclusion that, we don't. It does mean that, we need to document what intree metrics for these operations are and what are the CSI equivalents. So I have just noted that point and removed the need for new metrics.

Reply:
Can you also document those metrics in this section?

Reply:
done.